src/third_party/catch2/CMakeLists.txt

```diff
@@ -1,20 +1,17 @@
-# For more info on interface libraries,
-# see https://cmake.org/cmake/help/v3.10/command/add_library.html#object-libraries
-add_library(catch2_headers INTERFACE)
-target_include_directories(catch2_headers
-    INTERFACE
-        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-        $<INSTALL_INTERFACE:include>
-)
 
-# For more info on object libraries,
-# see https://cmake.org/cmake/help/v3.10/command/add_library.html#object-libraries
-add_library(catch2 OBJECT ${CMAKE_CURRENT_SOURCE_DIR}/main.cpp)
+add_library(catch2 SHARED ${CMAKE_CURRENT_SOURCE_DIR}/main.cpp)
```

Comment:
treating catch2 just like any other library got around some awkwardness with object librareis not being able to be added to target includes or somesuch and it also gets rid of a cmake_policy warning which is nice. This said I'm not sure of the reasons why catch2 wanted to be an object library (I read the link) so depending on those reasons portions of the linking with catch2 in the `genny_subdir` function may need to be changed.

Reply:
Along the same line, Does catch2 need a `main.cpp` that just includes the header file? If not, we can make catch2 an INTERFACE library and simplify a lot of things.

Reply:
Uhhhhh, so the catch2 library does main functions in a weird way that I haven't looked into too deeply. I had it as an object library so that each test would have its main function included in its binary.

That said. I honestly don't believe this file should be a thing. `catch2` should be an interface library and each test executable should have its own `test_foo.cpp` that looks like this main and whatever else we care about. (I suspect there is some sort of global teardown/setup in catch2 that might differ per unit.)

Reply:
It turned out to be straightforward and easy to treat catch2 just like a regular genny subdir so moved things around a bit and just used `genny_subdir` (now renamed to `CreateGennyTargets`).

Catch2 can't be *truly8 header-only because there needs to be exactly one cpp file in the test-targets that have the main defined (comes in by defining `CATCH_CONFIG_MAIN` before call to include catch2.hpp). Each genny test subdir could have its own main.cpp that does this but that seems like needless boilerplate; the main object-code may as well live with the catcch2 lib.